airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -467,12 +489,37 @@ class IncrementalStream(Stream, ABC):
     # Flag which enable/disable chunked read in read_chunked method
     # False -> chunk size is max (only one slice), True -> chunk_size is 30 days
     need_chunk = True
+    state_checkpoint_interval = 500
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return self.updated_at_field
 
     @property
     @abstractmethod
     def updated_at_field(self):
         """Name of the field associated with the state"""
 
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ):
+        if stream_state:
+            self.state = stream_state
```

Comment:
Set initial `state` which is got from `connector_state` mapping, passed from `AbstractSource` `_read_incremental` method.